server/src/chaos/models.rs

```diff
@@ -1,41 +1,573 @@
+            }
+        }
+        Ok(())
+    }
+
+    fn make_channel_type(channel_type: &ChannelTypeSQL) -> chaos_proto::chaos::Channel_Type {
+        use chaos_proto::chaos::Channel_Type;
+        match channel_type {
+            ChannelTypeSQL::Title => Channel_Type::title,
+            ChannelTypeSQL::Beacon => Channel_Type::beacon,
+            ChannelTypeSQL::Twitter => Channel_Type::twitter,
+            ChannelTypeSQL::Notification => Channel_Type::notification,
+            ChannelTypeSQL::Sms => Channel_Type::sms,
+            ChannelTypeSQL::Facebook => Channel_Type::facebook,
+            ChannelTypeSQL::Email => Channel_Type::email,
+            ChannelTypeSQL::Mobile => Channel_Type::mobile,
+            ChannelTypeSQL::Web => Channel_Type::web,
+        }
+    }
+
+    fn update_application_pattern(
+        application_pattern_set: &mut HashSet<Uid>,
+        row: &ChaosDisruption,
+        impact: &mut chaos_proto::chaos::Impact,
+    ) -> Result<(), Error> {
+        if let Some(pattern_id) = row.pattern_id {
+            if application_pattern_set.insert(pattern_id) {
+                let mut pattern = chaos_proto::chaos::Pattern::new();
+                if let Some(start_date) = row.pattern_start_date {
+                    pattern.set_start_date(u32::try_from(start_date.and_hms(0, 0, 0).timestamp())?)
+                }
+                if let Some(end_date) = row.pattern_end_date {
+                    pattern.set_end_date(u32::try_from(end_date.and_hms(0, 0, 0).timestamp())?)
+                }
+                // time_slot_begin && time_slot_end have always the same size
+                // even after filter_map
+                // thanks to the sql query
+                let time_slots_iter = row
+                    .time_slot_begin
+                    .iter()
+                    .flatten()
+                    .zip(row.time_slot_end.iter().flatten());
+
+                for (begin, end) in time_slots_iter {
+                    let mut time_slot = chaos_proto::chaos::TimeSlot::new();
+                    time_slot.set_begin(begin.num_seconds_from_midnight());
+                    time_slot.set_end(end.num_seconds_from_midnight());
+                    pattern.time_slots.push(time_slot);
+                }
+                impact.application_patterns.push(pattern);
+            }
+        }
+        Ok(())
+    }
+
+    fn update_pt_objects(
+        pt_object_set: &mut HashSet<String>,
+        row: &ChaosDisruption,
+        impact: &mut chaos_proto::chaos::Impact,
+    ) -> Result<(), Error> {
+        let id = if let Some(id) = &row.ptobject_uri {
+            id.clone()
+        } else {
+            bail!("PtObject has no uri");
+        };
+
+        let pt_object_type = row.ptobject_type.clone();
+
+        // Early exit if we already pushed a pt_object in impacts.informed_entities[]
+        // except for Line/rail section (they can be updated)
+        if !pt_object_set.insert(id.clone())
```

Comment:
```suggestion
        if !pt_object_set.contains(id.clone())
```
A contains seems more appropriate here, you don't want to add something to pt_object_set on early exit